apps/agentstack-ui/src/modules/connectors/api/index.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2025 © BeeAI a Series of LF Projects, LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { api } from '#api/index.ts';
+import { ensureData } from '#api/utils.ts';
+import { CreateConnectorRequest } from './types';
+import { BASE_URL } from '#utils/constants.ts';
+
+export async function createConnector(body: CreateConnectorRequest) {
+  const response = await api.POST('/api/v1/connectors', { body });
+  return ensureData(response);
+}
+
+export async function deleteConnector(connectorId: string) {
+  const response = await api.DELETE('/api/v1/connectors/{connector_id}', {
+    params: { path: { connector_id: connectorId } },
+  });
+  return ensureData(response);
+}
+
+export async function disconnectConnector(connectorId: string) {
+  const response = await api.POST('/api/v1/connectors/{connector_id}/disconnect', {
+    params: { path: { connector_id: connectorId } },
+  });
+  return ensureData(response);
+}
+
+export async function connectConnector(connectorId: string) {
+  const response = await api.POST('/api/v1/connectors/{connector_id}/connect', {
+    params: { path: { connector_id: connectorId } },
+    body: {
+      redirect_url: `${BASE_URL}/oauth-callback`,
+    },
+  });
+
+  return ensureData(response) as AuthRequired;
+}
+
+interface AuthRequired {
+  id: string;
+  url: string;
+  state: 'auth_required';
+  auth_request: {
+    authorization_endpoint: string;
+    type: string;
+  };
+}
+
+interface Connected {
+  id: string;
+  url: string;
+  state: 'connected';
+}
+
+interface Disconnected {
+  id: string;
+  url: string;
+  state: 'disconnected';
+}
+
+interface Created {
+  id: string;
+  url: string;
+  state: 'created';
+}
+
+// TODO: move
+type ListConnectorsResponse = {
```

Comment:
This whole typing here is jus temporary workaround  before properly generated in the openapi schema.